### PR TITLE
Game Class Refactor

### LINKED ISSUE #46: Add game class modifications

### PR DESCRIPTION
**Pull request by Manason**
Refactored Game class as well as updated the controller and view to work with the refactored Game class. Tested game and works fine.

fixes #46

### CODE REVIEW

File: src/main/java/models/Game.java [hunk 1]
```diff
@@ -9,92 +9,31 @@
...
-    public java.util.List<Card> deck = new ArrayList<>();
-
-    public java.util.List<java.util.List<Card>> cols = new ArrayList<>(4);
+    //deck is private as only Game should be drawing from deck, score and cols need to be accessed by the view however
+    private Deck deck;
+    public Columns cols;
+    public int score;
 
 
     public Game(){
```
Comment (loughrao) [inline src/main/java/models/Game.java]: What is the reason for having resetGame, if it is the same function as Game()? It seems like we could just put it all in game() and call game() when we need to reset. I could be misunderstanding how constructors work but I thought this was an option.

**Comment by loughrao** [comment]
What is the reason for having resetGame, if it is the same function as Game()? It seems like we could just put it all in game() and call game() when we need to reset. I could be misunderstanding how constructors work but I thought this was an option.

File: src/main/java/models/Game.java (hunk 1 above)
Comment (Manason) [reply src/main/java/models/Game.java]: You are correct. I left it as a separate function as there may be data that we want to transfer between rounds, which allows us to call resetGame() and reset pertinent data while keeping what we want to. Doing this in the constructor would be destructive.

Comment (loughrao) [reply src/main/java/models/Game.java]: Ah okay, I didn't realize we didn't have that functionality with the constructor.

**Comment by Manason** [comment]
tacking on issue #49 to this with the latest commits.

**Comment by porterga** [comment]
Good job thinking ahead on the possibility of needed to save data from previous games. Looks good, although it won't work with the aces feature, but that is more of an issue with the way that the aces feature is coded.

**Pull request merged**